src/App.js

```diff
@@ -95,10 +95,19 @@ class App extends Component {
         })
     }
 
-    onMovableActive = index => e => this.setState({
+    onMovableActive = index => e =>{ 
+      this.setState({
         activeElementIndex: index
     })
-    
+  }
+    editOnchange = index => e =>{
+      const {elements} = this.state
+      elements[index].props.children = e.target.value
```

Comment:
Please try to directly avoid changing a nested Array & Object property. Changing a nested property or element will result in unexpected results because of Reference sharing. In the above code, you already modified the state even before executing setState.

The code should be something like this

```js
// by doing this we are creating a new array and then modifing that array
const newElements = [...elements]
newElements[index] = {
    ...newElements[index],
    props: {
        ...newElements[index].props,
        children: e.target.value
    }
}

this.setState({
    elements: newElements
}) 
```
Please read this to have a better picture of why value immutability must be important https://github.com/getify/Functional-Light-JS/blob/master/manuscript/ch6.md/#chapter-6-value-immutability